src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -584,6 +585,20 @@ impl MonitorHandle {
             .with_info(&self.proxy, |_, info| info.scale_factor)
             .unwrap_or(1)
     }
+
+    #[inline]
+    pub fn video_modes(&self) -> impl Iterator<Item = VideoMode>
+    {
+        self.mgr
+            .with_info(&self.proxy, |_, info| info.modes.clone())
+            .unwrap_or(vec![])
+            .into_iter()
+            .map(|x| VideoMode {
+                dimensions: (x.dimensions.0 as u32, x.dimensions.1 as u32),
+                refresh_rate: (x.refresh_rate as f32 / 1000.0 + 0.5) as u16,
```

Comment:
We're receiving the refresh rate from Wayland in millihertz, so we must divide it by a thousand to get it in hertz. We then effectively `ceil` the float before casting it into an integer, but I suppose we could've used `f32::ceil` here also.

Reply:
In all, I'm not sure it is the right approach to represent refresh rates as `u16` in Hertz. There are displays whose refresh rate is not an integer in this unit, no?

Reply:
The refresh rate is usually not an exact integer, but very close (e.g. 59.996 Hz). We could report the display refresh rate as a float, but Windows and iOS only report it as an integer, so it might be misleading to return a float on these platforms. I assume this is what you meant?

Reply:
I don't get what would be misleading about using a float while some platform always return integer values, could you elaborate on that?

We already have precedent with the DPI factor values, which are always integer on Wayland for example.

Reply:
To elaborate, I feel like that returning a float here implies that this is an exact refresh rate, when in fact it's truncated or rounded (on at least two platforms). Returning an integer type does not have this implication to me.

If we look at other windowing libraries, there's precedent of using integers to represent the refresh rate in SDL and GLFW. I don't feel particularly strongly about this either way, so I'm fine with changing this to return floats if we all agree on that.

Reply:
I personally have the other expectation, imo an integer refresh rate suggests that it is exact...

Anyway, I don't know much about which uses an app would have for the refresh rate. If rounding it doe snot pose any functionality issue, then I'm fine with it.

Reply:
Having slept on it, I think we should leave it as a rounded integer, so at least the behaviour is consistent across platforms. If the user requires an exact value, there's not much point in us returning an exact value on some platforms, and a rounded value on others. Furthermore, changing this to a floating point value would complicate the implementation (though slightly) for `VideoMode` as we would now have to implement `Hash` and `Eq` by hand. If there later appears an use-case where an exact value is required, we can revisit this. For now, perhaps we can document it in the function description that the returned value is an approximation?